src/components/fingerprint/index.js

```diff
@@ -0,0 +1,100 @@
+module.exports = {
+    template: require('fingerprint.html'),
+    data: function() {
+        return {
+	    width: 512,
+	    height: 512,
+	    stream: null,
+        };
+    },
+    props: ['fingerprint', 'friendname', 'context'],
+    created: function() {
+        console.debug('Fingerprint module created!');
+    },
+
+    methods: {
+        close: function() {
+	    if (this.stream != null)
+		this.stream.getVideoTracks()[0].stop();
+            this.$emit("hide-fingerprint");
+        },
+
+	scanQRCode: function() {
+	    console.log("Scan QR code");
+	    if (navigator.mediaDevices && navigator.mediaDevices.getUserMedia) {
+		var that = this;
+		navigator.mediaDevices.getUserMedia({ video: true }).then(function(stream) {
+		    that.stream = stream;
+		    setTimeout(() => {
+			var video = document.getElementById('video');
+			video.srcObject = stream;
+			video.play()
+			that.takeSnapshot(60)
+		    }, 100);
+		}).catch(function(error) {
+		    alert("Is webcam connected?");
+		    console.error(error);
+		    that.stream = null;
+		});
+	    }
+	},
+
+	takeSnapshot: function(attemptsLeft) {
+	    var canvas = document.createElement('canvas');
+	    canvas.width = 512;
+	    canvas.height = 512;
+	    var video = document.getElementById('video');
+	    var vctx = canvas.getContext('2d');
+	    vctx.drawImage(video, 0, 0, this.width, this.height);
+	    this.processSnapshot(attemptsLeft, vctx);
+	},
+
+	processSnapshot: function(attemptsLeft, vctx) {
+	    var pixels = this.convertCanvasToPixels(vctx)
+	    try {
+		var scanned = peergos.shared.fingerprint.FingerPrint.decodeFromPixels(pixels, this.width, this.height);
+		this.stream.getVideoTracks()[0].stop();
+		this.stream = null;
+		if (this.fingerprint.matches(scanned)) {
+		    alert("Friend successfully verified!");
+		} else
+		    alert("non matching fingerprint!!");
```

Comment:
Change this to something like "QR code did not match this person's identity on Peergos. Are you sure this person is who they say they are?"

We don't mention fingerprints anywhere else.